File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.swift
func solution(_ n:Int, _ m:Int) -> [Int] {
    return [gcd(n, m), n*m/gcd(n,m)]
}

func gcd(_ x: Int, _ y: Int) -> Int {
    var a = x
    var b = y
    var c = 0
    while true {
        c = b
        b = a%b
        a = c
        if b == 0 {
            return c
        }
    }
}

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def gcd(a,b):
    x = max([a,b])
    y = min([a,b])

    while True:
        x, y = y, x%y
        if y == 0:
            return x
        
def lcm(c,d):
    lcm = c * d // gcd(c,d)
    return lcm

a, b = map(int, input().split())

print(gcd(a, b))
print(lcm(a, b))